docs: improve documentation (DSP-1693)

irinaschubert opened the pull request:
resolves DSP-1693

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -1,15 +1,17 @@
-philosophical sense.
+specialist, data librarian, or a digital humanities scholar in a data modeling notation_".
+
+In this section, we will describe one of the notations that is used by dsp-tools to create a data model in the DSP
```
Personally, I'm not overly fond of those "we" clauses. To me that sounds too narrative and less instruction-like than e.g. a passive construction. (But that's cosmetics)

irinaschubert replied:
I totally agree with you. But the whole file is written like this (and in general, in a very informal/spoken style). I leave it as is for the moment. Otherwise we would have to rewrite all the files from scratch.

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -67,44 +68,45 @@ As you can see in the example below, you can have more then one prefix too. In t
 - users
 
-So a simple example definition of the "project" object could look like this:
+So, a simple example definition of the "project" object could look like this:
```
Again, "So, ..." sounds more like an oral narrative than a software user manual

irinaschubert replied:
Same here, I don't want to change the style - otherwise I would have to rewrite everything.

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -212,17 +214,16 @@ example a classification of disciplines in the Humanities might look like follow
 characterized by a subelement of "Vocal music". However a search for "Masses" would retrun only works that 
 have been characterized as such. The number of hierarchy levels is not limited, but for practical reasons
-should not exceed 3-4 levels 
+should not exceed 3-4 levels.
```
```suggestion
it should not exceed 3-4 levels.
```

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -113,15 +115,15 @@ At that point we will go through all of this step by step and take a more in dep
+This is a short name (string) for the project. It's meant to be like a nickname. If the name of the project is e.g. 
 "Albus Percival Wulfric Dumbledore", then the shortname for it could be "Albi". It should be in the form of a
 [xsd:NCNAME](https://www.w3.org/TR/xmlschema11-2/#NCName), that is a name without blanks and special characters like
 ":", ";", "&", "%" etc., but "-" and "_" are allowed
```
shouldn't those be in back-ticks too? (`:` etc)

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -250,194 +251,201 @@ A resource can be assigned to a taxonomic node within its properties. So a resou
 different categories at the same time (e.g. an image which depicts several categories at the same time). In these cases,
-a cardinality &gt; 1 allows to add multiple attributes
-of the same time. See further below the description of the [cardinalities](#cardinalities)
+a cardinality &gt; 1 allows to add multiple attributes of the same time. See further below the description of the
```
do characters like `<` need to be escaped as html entities in markdown?  
In any case, I'd either have it in back ticks (`cardinality > 1`) or write it out ("cardinality greater than one")

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -250,194 +251,201 @@ A resource can be assigned to a taxonomic node within its properties. So a resou
-  
-Each list must have exactely one root node which has the same form bu denotes the list itself.
 
 Here is an example on how to build a taxonomic structure with the help of JSON:
```
is it "with the help of", not rather "in" JSON?

irinaschubert replied:
absolutely ;)

BalduinLandolt commented on docs/dsp-tools-create.md:
```diff
@@ -250,194 +251,201 @@ A resource can be assigned to a taxonomic node within its properties. So a resou
-          }
-        ]
-      },
-      {...},{...}
```
You systematically removed those `...`s. I'm wondering however if they aren't helpful to users, as they suggest that something needs to go there; and if the user doesn't fill it in, the editor will yell at them. (Counts for all the other instances as well.)

irinaschubert replied:
I see your point, I changed it back to the version with three dots.

BalduinLandolt commented on docs/index.md:
```diff
@@ -1,210 +1,226 @@
-- _"-v" | "--verbose"_: Print out some information about progress
+The following options are available:
+
+- -s | --server _server_: URL of the DSP server (default: localhost:3333)
+- -u | --user _username_: username used for authentication with the DSP API (default: root@example.com)
+- -p | --password _password_: password used for authentication with the DSP API (default: test)
+- -V | --validate: If set, only the validation of the JSON file is performed.
+- -l | --lists: If set, only the lists are created using a [simplified schema](./dsp-tools-create.md#lists). Please note
+  that in this case the project must already exist.
+- -v | --verbose: If set, some information about the progress is printed to the console.
```
I suggest using backticks for the actual flags (`-s` etc.)

BalduinLandolt commented on docs/index.md:
```diff
@@ -1,210 +1,226 @@
-The above command line supports the following options:
+The following options are available:
+
+- -s | --server _server_: URL of the DSP server (default: localhost:3333)
+- -u | --user _username_: username used for authentication with the DSP API (default: root@example.com)
+- -p | --password _password_: password used for authentication with the DSP API (default: test)
+- -P | --project _shortcode_ | _shortname_ | _iri_: shortcode, shortname or
+  [IRI](https://en.wikipedia.org/wiki/Internationalized_Resource_Identifier) of the project
+- -v | --verbose: If set, some information about the progress is printed to the console.
```
as above

BalduinLandolt commented on docs/index.md:
```diff
@@ -1,210 +1,226 @@
-- _"-i dirpath" | "--imgdir"dirpath"_: Path to the directory where the bitstream objects are stored [default: "."].
-- _"-s sipiserver" | "--sipi sipiserver"_: URL of the SIPI IIIF-server [default: http://0.0.0.0:1024]
-- _"xmlfile"_: Path to xml-File containing the data.
+- -s | --server _server_: URL of the DSP server (default: localhost:3333)
+- -u | --user _username_: username used for authentication with the DSP API (default: root@example.com)
+- -p | --password _password_: password used for authentication with the DSP API (default: test)
+- -i | --imgdir _dirpath_: path to the directory where the bitstream objects are stored (default: .)
+- -S | --sipi _SIPIserver_: URL of the SIPI IIIF server (default: http://0.0.0.0:1024)
```
as above

BalduinLandolt commented on docs/index.md:
```diff
@@ -1,210 +1,226 @@
+The following options are available:
 
-## Makefile
+- -S | --sheet _sheetname_: name of the Excel worksheet to use (default: Tabelle1)
+- -s | --shortcode _shortcode_: shortcode of the project (required)
+- -l | --listname _listname_: name to be used for the list and the list definition file (required)
+- -L | --label _label_: label to be used for the list (required)
+- -x | --lang _lang_: language used for the list labels and commentaries (default: en)
+- -v | --verbose: If set, some information about the progress is printed to the console.
```
as above

BalduinLandolt reviewed (approved):
good job, that improves the documentations a lot!  
I added some comments and suggestions, but that's all cosmetics, feel free to merge it as is, if you prefer.

Another thought that came to my mind: Some of the documentation pages are extremely long. It might be worth considering to split those into multiple files to have them as sub-pages of the page in the TOC

irinaschubert commented:
@BalduinLandolt , thank you for the feedback, I considered almost everything. I also split the index page into several sub pages ("Overview", "Usage" and "Information for developers"). I totally agree that it would be preferable to also split the two other long pages but I think they should undergo an intense revision anyway, so this could be done at that point.
